### ISSUE TITLE
Target size: false positive on overlapping elements

### ISSUE THREAD

**Issue by andrewhick** [dequelabs/axe-core#4805]
### Product

axe-core

### Product Version

axe-core 4.10.3

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation
Axe should not fail two overlapping elements for [2.5.8 Target Size](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum), where the centres of both elements are more than 24px away from each other.

#### Actual
Axe fails two elements whose centres are more than 24px apart.

In this case a `button` is embedded within a link. The link is split vertically into three equally-sized sections, and the final section includes a button less than 24px high.

#### How to Reproduce
An example can be found on [https://andrewhick.com/accessibility/failures.html#targetsize-axefalsepositive](https://andrewhick.com/accessibility/failures.html#targetsize-axefalsepositive) under Target Size and Axe False Positive.

Code:

```html
<a href="#">
	<div class="submenu" style="max-width: 100%; height: 32px; border: 2px solid #36d;">
		<div style="width: 33%; float: left;">Section 1</div>
		<div style="width: 33%; float: left;">Section 2</div>
		<div style="width: 33%; float: left;">Section 3 <button style="font-size: 12px;">button</button></div>
	</div>
</a>
```

#### Additional context
This was originally detected on a "real life" website displaying similar properties. This issue looks similar to #4224 but that issue focuses on the fact that the two links in that example have the same destination. Happy to discuss further.

**Comment by WilcoFiers** [comment]
@andrewhick This is intended to fail WCAG, there is a very similar example in [WCAG's Target Size understanding document](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html#:~:text=it%20fails%20the%20criterion%2C%20as%20the%20imaginary%20circle%20with%20a%2024%20CSS%20pixel%20diameter%20we%20draw%20over%20the%20small%20target%20will%20intersect%20the%20large%20target%20itself).

The spacing exception says this:

> **Spacing**: Undersized targets (those less than 24 by 24 CSS pixels) are positioned so that if a 24 CSS pixel diameter circle is centered on the [bounding box](https://www.w3.org/TR/2023/REC-WCAG22-20231005/#dfn-bounding-boxes) of each, the circles do not intersect another target or the circle for another undersized target;

For each undersized target, there are two requirements:
1. The circle does not intersect other targets; and
2. The circle does not intersect the circle of other undersized targets

In your example, you're right that the button's 24px circle doesn't intersect the 24px circle of the link (condition 2), although since the link isn't an undersized element that's not required. What it does do is intersect with the target of the link. It's not allowed to do that. It is not possible for a target that's on top of / enclosed by another target to have a diameter of less than 24px.
